core/src/main/scala/com/pingcap/tispark/telemetry/TiSparkTeleInfo.scala

```diff
@@ -93,11 +94,22 @@ object TiSparkTeleInfo {
       val conf: TiConfiguration = new TiConfiguration
       TiUtil.sparkConfToTiConfWithoutPD(SparkSession.active.sparkContext.getConf, conf)
 
+      var pd_uri: URI = null;
       if (conf.isTlsEnable) {
-        val url = "https://" + pd_address.get + urlPattern
+        pd_uri = new URI(s"https://${pd_address.get}$urlPattern")
+      } else {
+        pd_uri = new URI(s"http://${pd_address.get}$urlPattern")
+      }
+
+      if (conf.getHostMapping != null) {
+        pd_uri = conf.getHostMapping.getMappedURI(pd_uri)
+      }
+
+      if (conf.isTlsEnable) {
+        val url = pd_uri.toString
         resp = httpClient.getHttpsWithTiConfiguration(url, conf)
       } else {
-        val url = "http://" + pd_address.get + urlPattern
+        val url = pd_uri.toString
```

Comment:
Can we remove the val url here and above, just use pd_uri.toString.
Since url is only used once, and defining url doesn't increase the readability.

Reply:
Yes, It is significant to make the code more readable. I will delete the unuseful variable.